sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -1707,6 +1732,17 @@ mod checked {
         }
     }
 
+    fn unwrap_type_tag_load(
+        protocol_config: &ProtocolConfig,
+        ty: Result<Type, ExecutionError>,
+    ) -> Result<Type, ExecutionError> {
+        if ty.is_err() && !protocol_config.type_tags_in_object_runtime() {
```

Comment:
This should never happen, but just to be paranoid, panic in case it does and we're before this feature is enabled